src/current/_includes/v24.1/sql/role-subject-option.md

```diff
@@ -0,0 +1 @@
+You can associate an [X.509](https://en.wikipedia.org/wiki/X.509) certificate's Subject with a [role]({% link {{ page.version.version }}/security-reference/authorization.md %}#roles) as shown below. Note that the Subject fields in the certificate have to be an exact match with what you pass in via the SQL statement.
```

Comment:
@bdarnell or @souravcrl do you know the answer to this question?

I think answer is probably "yes they need to be in the same order", based on the following (but I defer to Ben and Sourav since this is just me code spelunking with little understanding):

I looked around the code and found that [`ParseDNFromCertificate`](https://github.com/cockroachdb/cockroach/blob/master/pkg/security/distinguishedname/parse.go#L52) has a comment that says

> ... It retains the sequence of fields as provided in the certificate subject ...

I also looked at some [code the LDAP library uses for checking if DNs are equal](https://github.com/go-ldap/ldap/blob/56e5759aadc48129a33935b9de05b17151c1172e/dn.go#L309C4-L309C9) and it looks like it iterates over a list in order.  The comment also says

> Returns true if they have the same number of relative distinguished names and corresponding relative distinguished names (by position) are the same

Reply:
Thanks, I've updated to add another sentence that says the order must match